doc/tutorial/riscv-virtual-memory.rst

```diff
@@ -0,0 +1,413 @@
+
+Now that we are using the pagetable, let's diplay its content.
+If you have tried to decode the pagetable manually, it's time to check your results.
+We can use the ``ptd`` command to dump the currently used pagetable:
+
+.. code:: msim
+
+    [msim] cpu0 ptd
+    satp 0x800a0000 [ Mode: Sv32 ASID: 0 PPN: 0x0a0000 (Physical address: 0x0a0000000) ]
+    0x800: [ PPN: 0x080000 RSW: 00 -AG- XWRV ] [ Megapage ]
+    0x900: [ PPN: 0x090000 RSW: 00 --G- -WRV ] [ Megapage ]
+    0xa00: [ PPN: 0x0a0000 RSW: 00 --G- -WRV ] [ Megapage ]
+    0xb00: [ PPN: 0x0a0001 RSW: 00 ---- ---V ]
+      0x000: [ PPN: 0x0c0000 RSW: 00 ---- XWRV ]
+      0x008: [ PPN: 0x0c0000 RSW: 00 ---- --RV ]
+      0x00c: [ PPN: 0x0c0000 RSW: 00 --G- XWRV ]
+      0x010: [ PPN: 0x0c0000 RSW: 00 ---U XWRV ]
+
+This command first displays the content of ``satp``
+so we can check which pagetable we are working with.
+Then it traverses the pagetable, displaying all valid PTEs.
+For each non-leaf PTE it descends into the second-level pagetable
+and displays its valid PTEs.
+
+Each line thus corresponds to one PTE.
+It starts with the address offset of this particular PTE in its page,
+then it displays the stored PPN and RSW bits.
+It ends with the individual bitfields ``DAGU XWRV``.
+A corresponding letter is displayed if this bit is ``1``,
+a dash is present instead if this bit is ``0``.
+PTEs representing a megapage are denoted as such,
+second level PTEs are indented with two spaces.
+
+.. quiz::
+
+    What do the individual letters in ``DAGU XWRV`` stand for?
+
+    .. collapse:: Hint
+
+        Look at the `RISC-V Privileged Specification <https://github.com/riscv/riscv-isa-manual/releases/download/20240411/priv-isa-asciidoc.pdf>`__
+        Chaper 10.3. Sv32: Page-Based 32-bit Virtual-Memory Systems.
+
+    .. collapse:: Solution
+
+        - **D**\ irty
+        - **A**\ ccessed
+        - **G**\ lobal
+        - **U**\ ser
+        - e\ **X**\ ecute
+        - **W**\ rite
+        - **R**\ ead
+        - **V**\ alid
+
+If you want to display invalid PTEs in addition to the valid ones,
+you can use the verbose flag:
+
+.. code:: msim
+
+    [msim] cpu0 ptd v
+    satp 0x800a0000 [ Mode: Sv32 ASID: 0 PPN: 0x0a0000 (Physical address: 0x0a0000000) ]
+    0x800: [ PPN: 0x080000 RSW: 00 -AG- XWRV ] [ Megapage ]
+    0x900: [ PPN: 0x090000 RSW: 00 --G- -WRV ] [ Megapage ]
+    0xa00: [ PPN: 0x0a0000 RSW: 00 --G- -WRV ] [ Megapage ]
+    0xb00: [ PPN: 0x0a0001 RSW: 00 ---- ---V ]
+      0x000: [ PPN: 0x0c0000 RSW: 00 ---- XWRV ]
+      0x004: [ PPN: 0x0c0000 RSW: 00 ---- XWR- ]
+      0x008: [ PPN: 0x0c0000 RSW: 00 ---- --RV ]
+      0x00c: [ PPN: 0x0c0000 RSW: 00 --G- XWRV ]
+      0x010: [ PPN: 0x0c0000 RSW: 00 ---U XWRV ]
+
+This way, all non-zero PTEs are displayed (and indeed, there are 9 of them ;-) ).
+
+If you don't want to dump the content of the currently active pagetable,
+but would rather specify it by its (physical) address,
+you can use the ``sptd`` command.
```

Comment:
Perhaps add that the `s` stands for simulated to easier remembering of the command?